mesonbuild/coredata.py

```diff
@@ -1003,11 +1005,12 @@ def set_default_options(self, default_options: T.MutableMapping[OptionKey, str],
             # not differ between them even when they are valid for both.
             if subproject and k.is_builtin() and self.options[k.evolve(subproject='', machine=MachineChoice.HOST)].yielding:
                 continue
-            # Skip compiler and backend options, they are handled when
+            # Skip base, compiler, and backend options, they are handled when
             # adding languages and setting backend.
-            if k.type in {OptionType.COMPILER, OptionType.BACKEND}:
+            if k.type == OptionType.BASE and k in base_options:
+                # set_options will report unknown base options
                 continue
-            if k.type == OptionType.BASE and env.first_invocation:
+            if k.type in {OptionType.COMPILER, OptionType.BACKEND}:
```

Comment:
Was there a reason to switch the order of these checks?

Reply:
This is not really an ordre switch... There are two things here. 1- I reverted my previous commit because it was wrong to pass base options to set_options.  2- I added a condition to pass *unknown* base options to set_options, only to ensure it raises an error. The fact that the order is switched is a coincidence, and doesn't matter.

Reply:
In current main, we first check `k in {compiler, backend}` then we check `k == base`. In you pacthc you've switched the order of the checks. Is there a reason for that?

Reply:
No. I occured just because I removed one line in one commit, and added the other in the other commit. I fixed it.

Reply:
Thanks! LGTM now